app/src/main/java/com/example/admin/learningandroid/threads/MakeAPICallActivity.java

```diff
@@ -0,0 +1,188 @@
+package com.example.admin.learningandroid.threads;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.os.AsyncTask;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.example.admin.learningandroid.R;
+import com.example.admin.learningandroid.entity.WeatherAPIModelClass;
+import com.example.admin.learningandroid.layouts.RecyclerViewAdapter;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.zip.Inflater;
+
+
+import static com.example.admin.learningandroid.Constants.WEATHER_URL;
+
+/* this activity class is used to implement a network call without using library
+ * Here an api call to a http url is made and JSON data is retrieved
+ * The JSON data is converted to JSON objects manually with the help of JSON object
+ * and JSON array classes*/
+public class MakeAPICallActivity extends AppCompatActivity {
+    Context mContext;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.make_api_call_layout);
+        APICallAsyncTask apiCallAsyncTask = new APICallAsyncTask();
+        apiCallAsyncTask.execute();
+    }
+
+
+    public String getResponse() throws IOException {
+
+        URL weatherURL = new URL(WEATHER_URL);
```

Comment:
Can there be on Entity for weather object and one adapter since the contents are the same